### PR TITLE
[JSX] Add CSSGrid component type

### PR DESCRIPTION
**Pull request by driusan**
This adds a CSSGrid component type. A CSSGrid aligns
a number of cards (passed as a prop) in a grid. The
component takes care of converting the values passed
to it into cards with an appropriate title, size
and positioning.

### CODE REVIEW

**Comment by zaliqarosli** [comment]
@driusan i am wondering why CSSGrid over Flexbox?

**Comment by driusan** [comment]
@zaliqarosli flexbox only works in 1 dimension. A grid is like a 2 dimensional flexbox.

**Comment by driusan** [comment]
I'd say "see https://www.webdesignerdepot.com/2018/09/grid-vs-flexbox-which-should-you-choose/" but the images in that link seem to be broken..

File: jsx/CSSGrid.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    gridTemplateColumns: '33% 33% 33%',
+    gridAutoFlow: 'row dense',
+    gridRowGap: '1em',
+    rowGap: '1em',
+  };
+
+  let cards = props.Cards.map((value, idx) => {
+    let cardID = 'card' + idx;
+
+    let style= {};
```
Comment (zaliqarosli) [inline jsx/CSSGrid.js]:
```suggestion
    let style = {};
```

File: jsx/CSSGrid.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    if (value.Height) {
+      style.gridRowEnd = 'span ' + value.Height;
+    }
+    if (value.Order) {
+      style.order = value.Order;
+    }
+
+    style.alignSelf = 'stretch';
+    return (<Card title={value.Title} id={cardID} key={cardID} style={style}>
```
Comment (zaliqarosli) [inline jsx/CSSGrid.js]:
```suggestion
    return (
      <Card title={value.Title} id={cardID} key={cardID} style={style}>
        {value.Content}
      </Card>
    );
```

File: jsx/CSSGrid.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      style.order = value.Order;
+    }
+
+    style.alignSelf = 'stretch';
+    return (
+      <Card title={value.Title} id={cardID} key={cardID} style={style}>
+        {value.Content}
+      </Card>
+    );
+      {value.Content}
```
Comment (zaliqarosli) [inline jsx/CSSGrid.js]:
```suggestion

```

File: jsx/CSSGrid.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      </Card>);
```
Comment (zaliqarosli) [inline jsx/CSSGrid.js]:
```suggestion

```

**Pull request merged**